app/design/frontend/base/default/template/algoliasearch/internals/configuration.phtml

```diff
@@ -257,6 +257,9 @@ $algoliaJsConfig = array(
 
 <script>
     window.algoliaConfig = <?php echo json_encode($algoliaJsConfig); ?>;
+    if (typeof algoliaHookAfterConfigLoad == 'function') {
+        window.algoliaConfig = algoliaHookAfterConfigLoad(algoliaConfig);
```

Comment:
Given the configuration is generated on the backend, I'm afraid it can be hard to debug if at a given point a user customizes in the backend and in the frontend with a hook.